oak_functions/lookup/src/lib.rs

```diff
@@ -183,19 +181,50 @@ where
         test_manager
     }
 
-    /// Updates the backing data that will be used by new `LookupData` instances.
-    pub fn update_data(&self, action: UpdateAction, new_data: Data) -> UpdateStatus {
+    pub fn update_start_and_finish(&self, new_data: Data) -> UpdateStatus {
         let mut data_builder = self.data_builder.lock();
+        match &data_builder.state {
+            BuilderState::Empty => {
+                data_builder.extend(new_data);
+                let next_data = data_builder.build();
+                let mut data = self.data.lock();
+                *data = Arc::new(next_data);
+                UpdateStatus::Finished
```

Comment:
Maybe worth a comment that this method needs to do the same things as update_start and update_finish (and must be manually kept in sync if they change)

Reply:
Fixed by calling start and finish with empty data (thanks for the comment :)